manifests/agent/install.pp

```diff
@@ -1,8 +1,13 @@
 # Install the puppet client installation
 class puppet::agent::install {
-
-  package { $puppet::params::client_package:
-    ensure => $::puppet::version,
+  if ($::puppet::params::package_provider) {
+    package { $puppet::params::client_package:
+      ensure   => $::puppet::version,
+      provider => $::puppet::params::package_provider,
+    }
+  } else {
+    package { $puppet::params::client_package:
+      ensure => $::puppet::version,
+    }
   }
```

Comment:
@ekohl I think I addressed this the way you had in mind, if not just let me know. I actually agree wholeheartedly about keeping the settings generic.